core/src/main/kotlin/org/opensearch/alerting/core/model/DataSources.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.core.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.ToXContentObject
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+data class DataSources(
+    /** Configures a custom query index name for the monitor. Creates a new index if index with given name not present.*/
+    val queryIndex: String? = null,
+
+    /** Configures a custom index to store findings for a monitor. Creates a new index if index with given name not present.
+     *  If index is pre-existing, mapping is updated*/
+    val findingsIndex: String? = null,
+
+    /** Configures a custom index to store alerts for a monitor. Creates a new index if index with given name not present.
+     *  If index is pre-existing, mapping is updated. */
+    val alertsIndex: String? = null,
+
+    /** Configures custom mappings by field type for query index.
+     * Custom query index mappings are configurable, only if a custom query index is configured too. */
+    val queryIndexMappingsByType: Map<String, Map<String, String>> = mapOf()
```

Comment:
If analyzer is the only type right now, we might want an enum set of what we support for query index mappings for when its open to other users and plugins. We might want a follow up item where with percolate queries, what other fields would make sense and restrict it to just those fields. When we open it up, to everyone, it might be a bad user experience to keep it open ended since it makes the usability harder for more novice users.

Reply:
Agree. Plan to open this up as part of Composite Monitors should cover the specific validations around which fields should have what analyzer .